src/brevitas_examples/common/learned_round/learned_round_optimizer.py

```diff
@@ -0,0 +1,820 @@
+        disable_quant: bool = False) -> None:
+    if disable_quant:
+        disable_quant_class = DisableEnableQuantization()
+        disable_quant_class.disable_act_quantization(model, False)
+        disable_quant_class.disable_param_quantization(model, False)
+        return_quant_tensor_state = disable_return_quant_tensor(model)
+
+    data_saver = DataSaverHook(
+        cache, store_inputs=store_inputs, store_output=store_output, keep_gpu=keep_gpu)
+    handle = module.register_forward_hook(data_saver, with_kwargs=True)
+    with torch.no_grad():
+        for inps in dataloader:
+            try:
+                model_forward(model, inps)
+            except StopFwdException:
+                pass
+    handle.remove()
+    if disable_quant:
+        disable_quant_class.enable_act_quantization(model, False)
+        disable_quant_class.enable_param_quantization(model, False)
+        restore_return_quant_tensor(model, return_quant_tensor_state)
+
+
+class LearnedRoundOptimizer:
+
+    def __init__(
+        self,
+        learned_round: LearnedRound,
+        learned_round_loss_class: Type[LearnedRoundLoss],
+        *,
+        optimizer_class: Type[Optimizer] = SignSGD,
+        scale_optimizer_class: Type[Optimizer] = SGD,
+        lr_scheduler_class: Optional[Type[LRScheduler]] = LinearLR,
+        optimizer_lr: float = 5e-3,
+        batch_size: float = 8,
+        iters: int = 200,
+        learn_scale: bool = False,
+        use_best_model: bool = True,
+        use_amp: bool = True,
+        amp_dtype: torch.dtype = torch.float16,
+        loss_scaling_factor: float = 1000.,
+        use_accelerate: bool = False,
+        learned_round_loss_kwargs: Optional[Dict] = None,
+        optimizer_kwargs: Optional[Dict] = None,
+        lr_scheduler_kwargs: Optional[Dict] = None,
+    ) -> None:
+        self.learned_round = learned_round
+        self.optimizer_class = optimizer_class
+        self.scale_optimizer_class = scale_optimizer_class
+        self.lr_scheduler_class = lr_scheduler_class
+        self.optimizer_lr = optimizer_lr
+        self.batch_size = batch_size
+        self.iters = iters
+        self.learn_scale = learn_scale
+        self.use_best_model = use_best_model
+        self.use_amp = use_amp
+        self.amp_dtype = amp_dtype
+        self.loss_scaling_factor = loss_scaling_factor
+        self.optimizer_kwargs = {} if optimizer_kwargs is None else optimizer_kwargs
+        self.lr_scheduler_kwargs = {} if lr_scheduler_kwargs is None else lr_scheduler_kwargs
+        self.lr_scheduler_kwargs["total_iters"] = self.iters
+
+        learned_round_loss_kwargs = {} if learned_round_loss_kwargs is None else learned_round_loss_kwargs
+        self.learned_round_loss_init = partial(
+            learned_round_loss_class, **learned_round_loss_kwargs)
+
+        # TODO: Remove once validated and expose the flag
+        # self.use_accelerate = use_accelerate
+        self.use_accelerate = False
```

Comment:
What does this mean

Reply:
I started the distributed training optimization. Maybe it's worth removing it and leave it for another PR.